### ISSUE TITLE
3RVX: sounds at windows startup

### ISSUE THREAD

**Issue by seb3343** [malensek/3RVX#8]
I have been using 3RVX on my Toshiba laptop with Win7 for the last few weeks and I love it. There is only one minor issue, at least on my laptop: when I start windows, the 3RVX sound display pops up 3 or 4 times (at about 10 sec intervals) and it plays the "volume change" sound (2 beeps). After 3 or 4 times, it stops doing it automatically. Have you ever seen this? Is there a fix?

Thanks

**Comment by malensek** [comment]
Hmm, this is strange. Are you able to run the latest beta version here:

http://matthew.malensek.net/projects/3RVX/3RVX3-B3.zip

If you can test it, does it exhibit the same behavior? The reason I'm assuming you're on v2 is that the volume change sound doesn't work in v3 yet.

**Comment by seb3343** [comment]
Thanks for the fast answer. 
Yes, I am using v 2.5
I downloaded the latest beta version but for some reason I can't install it. I get the windows warning window, click ok, and then nothing happens.

**Comment by jackwilsdon** [comment]
@seb3343 I agree the white icon is a bit of an issue with the default Classic skin, however changing skins to pix fixes it.

**Comment by seb3343** [comment]
Thank you but I am not sure we're talking about the same icons. I am talking about the very small notification icons that are to the right of the task bar, next to the clock. There are a bunch of them when we click on the arrow pointing upwards. 3RVX is one of them but it is invisible.

Anyway, this is not a major issue - my first question is more important.

**Comment by jackwilsdon** [comment]
I don't think the old version (2.5) is supported anymore. You'll have to wait until the Beta fixes any issues it has.

**Comment by malensek** [comment]
Hey @seb3343 , could you try out the latest beta (B4) and see if this version will launch correctly? As for the icon thing, I'll definitely get it fixed soon.

http://matthew.malensek.net/projects/3RVX/3RVX3-B4.zip

**Comment by seb3343** [comment]
Same issue - I get the windows warning and then nothing.

**Comment by malensek** [comment]
Ok, I have one more version for you to test, but there's a decent chance this'll do the same thing...

http://matthew.malensek.net/projects/3RVX/3RVX3-B5.zip

(B5 adds some new error messages instead of dying silently)

One of my priorities for the next beta is to add an option to dump a detailed debug log, and that should help pinpoint what's going on.

**Comment by seb3343** [comment]
Yes, exact same thing, and no error message.

Another thing that you might want to look into is to make sure 3RVX restarts after the computer goes to sleep. I noticed that with v 2.5, I need to restart it manually.

**Comment by Walkman100** [comment]
Are you sure you're allowing it through the windows warning?

**Comment by seb3343** [comment]
Yes, I am ... this is pretty basic!

**Comment by malensek** [comment]
Hey @seb3343 , sorry for taking so long to get back to you on this. I've created an executable that will log a bunch of diagnostic information to a file (3RVX_Log.txt in the same directory as the executable).

If you get some time, would you mind running this executable and then attach the log it produces? This'll give me an idea of where the program is crashing.

http://matthew.malensek.net/projects/3RVX/3RVX3-B7-pre2.zip

Thanks, and sorry again for the long delay. To answer your question about the sleep bug, I believe that is fixed in the new version.

**Comment by seb3343** [comment]
With this version, I get error message 258. See attached
![error 258](https://cloud.githubusercontent.com/assets/11361260/7161553/f2e05064-e38f-11e4-9f62-ffa10376e3ca.png)

**Comment by jackwilsdon** [comment]
Try deleting your `Settings.xml`, it's format may have changed.

**Comment by seb3343** [comment]
Deleting setting.xml from where? I don't see this file anywhere.

This is the log I get:
Starting up...
[wWinMain:72]
App directory: C:\Users\User\Desktop\3rvx
[init:110]
Initializing...
[Settings::Load:55]
Loading settings: C:\Users\User\Desktop\3rvx\Settings.xml

**Comment by jackwilsdon** [comment]
You'll need to get a copy of `Settings.xml` from somewhere else. Looks like the version @malensek uploaded doesn't include one. Download [this file](https://github.com/malensek/3RVX/blob/master/3RVX/Settings.xml) and save it in the same directory as the exe file. You'll need the `SettingsUI` program from a previous release too.

**Comment by malensek** [comment]
Whoops, I should've clarified... Jack is right, you will need the other files from the latest beta:

http://matthew.malensek.net/projects/3RVX/3RVX3-B6.zip

For the next release, I won't require a valid settings file anymore.

**Comment by seb3343** [comment]
No error message anymore but I don't think it does a full installation. It works, but it is not listed in Control Panel\Programs\Programs and Features. Not in windows - all programs either.

**Comment by malensek** [comment]
Ah yeah, this version is "portable" -- just unzip and run. Once I start doing final releases I'll make sure there's an installer version as well.

**Comment by seb3343** [comment]
OK, this one installs fine but the original issue remains (when I start windows, the 3RVX sound display pops up 3 or 4 times (at about 10 sec intervals) and it plays the "volume change" sound (2 beeps). After 3 or 4 times, it stops doing it automatically). This of course only occurs if I select "run on windows startup".

Also, with this new version, there doesn't seem to be an option to select the volume increment.

**Comment by malensek** [comment]
Ah, so the new version has the same issue. Would you mind posting the log file it produces when this bug occurs? I'd like to see if the volume is actually changing or if the program is just receiving events from Windows that make it think there was a change.

Thanks a lot for all your help with debugging this!!

**Comment by seb3343** [comment]
I just tried but (1) the log enabled version above doesn't install because there is no settings.xml file. (2) the other version with the settings.xml file installs fine but doesn't save a log ... so I tried combining them (log enabled exe file and settings.xml file from your latest post) but it doesn't automatically launch when I start windows. I would need a log-enabled version with the corresponding settings.xml file. Hope this helps.

**Comment by seb3343** [comment]
Just an update on this: I have installed the latest version 3.0 B8 and I still have this sound at startup issue with win7. Am I the only one?

**Comment by malensek** [comment]
I'll get a log-enabled installer version whipped up for the next beta (this should be way easier than jumping through all those hoops to get a log like before).

This seems like it could be an issue with the device being switched or toggled, or maybe something else is setting/resetting the volume... Something has to be creating these sound events. You mentioned that it doesn't happen when the program doesn't start with windows, but what happens if you immediately launch 3RVX manually (within that 30-40 sec interval)?

I hope to be focused more on bug fixes rather than new features for these next few versions.
